pkg/promote/promote.go

```
@@ -808,22 +807,27 @@ func (o *Options) ResolveChartRepositoryURL() (string, error) {
 
 // IsLocalChartRepository return true if the chart repository is blank or a local url
 func IsLocalChartRepository(repo string) bool {
-	repo = strings.TrimPrefix(repo, "https://")
-	repo = strings.TrimPrefix(repo, "http://")
-	repo = strings.TrimPrefix(repo, "https:")
-	repo = strings.TrimPrefix(repo, "http:")
-
-	repo = strings.TrimSuffix(repo, "/")
-	repo = strings.TrimSuffix(repo, ".jx")
-
+	repoURL, err := url.Parse(repo)
+	if err != nil {
```

errors can happen because of reasons other than it being an OCI registry address, can we return an error for those cases?

---

Do you really know that the repo is invalid in those cases? I don't.

---

I recall similar login in jx-helpers repo, will find out ...